java-checks/src/main/java/org/sonar/java/symexec/SymbolicEvaluator.java

```diff
@@ -68,20 +68,33 @@
 
 public class SymbolicEvaluator {
 
+  private static final int MAXIMAL_EXECUTION_STATE_COUNT = 65536;
+
   private final AssignedSymbolExtractor extractor = new AssignedSymbolExtractor();
 
   private final Map<Tree, SymbolicBooleanConstraint> result = new HashMap<>();
 
+  private int currentExecutionStateCount;
+  private int maximalExecutionStateCount;
+
+  public SymbolicEvaluator() {
+    maximalExecutionStateCount = MAXIMAL_EXECUTION_STATE_COUNT;
+  }
+
   public Map<Tree, SymbolicBooleanConstraint> evaluateMethod(ExecutionState state, MethodTree tree) {
     result.clear();
     if (tree.block() != null) {
-      evaluateStatement(ImmutableList.of(state), tree.block());
+      try {
+        evaluateStatement(ImmutableList.of(state), tree.block());
+      } catch (SymbolicExecutionException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)